ladybug/src/main/java/nl/nn/ibistesttool/MessageCapturer.java

```diff
@@ -0,0 +1,56 @@
+/*
+   Copyright 2021 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.ibistesttool;
+
+import java.io.OutputStream;
+import java.io.Writer;
+
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.stream.Message;
+import nl.nn.testtool.TestTool;
+
+public class MessageCapturer implements nl.nn.testtool.MessageCapturer {
+	private @Setter @Getter TestTool testTool;
+
+	@Override
+	public StreamingType getStreamingType(Object message) {
+		if (message instanceof Message) {
+			Message m = (Message)message;
+			if (m.requiresStream()) {
+				if (m.isBinary()) {
+					return StreamingType.BYTE_STREAM;
+				} else {
+					return StreamingType.CHARACTER_STREAM;
+				}
+			}
+		}
+		return StreamingType.NONE;
+	}
+
+	@Override
+	public <T> T toWriter(T message, Writer writer) {
+		((Message)message).captureCharacterStream(testTool.getMaxMessageLength(), writer);
+		return message;
+	}
+
+	@Override
+	public <T> T toOutputStream(T message, OutputStream outputStream) {
```

Comment:
Je zou aan de interface van toOutputStream een extra parameter van het type Consumer<String> toe kunnen voegen, waarmee de implementatie aan kan geven met welke character set de bytes (waarschijnlijk) gelezen zouden kunnen worden als characters.

Reply:
Wat mij betreft niet nodig, laat maar weten als jij het wel graag wil.

Reply:
Het is alleen bedoelt om aan Ladybug door te geven, zodat die er iets mee kan doen.

Reply:
Ik denk dat het verstandig is om wel te doen, daar ga je later profijt van hebben.